src/components/HistoricalView.tsx

```diff
@@ -1,11 +1,22 @@
-import { MurojaahType } from '@/api/module/murojaah/entity/murojaah';
+import { useEffect, useState } from 'react';
+import { History, MurojaahType } from '@/api/module/murojaah/entity';
+import { Index } from '@/api/module/murojaah/service';
 import { useAlert } from '@/context/AlertContext';
-import { Card, Props } from '@/components/Card';
+import { Card } from '@/components/Card';
 import { clsx } from 'clsx';
 
 export const HistoricalView = (): JSX.Element => {
+  const [data, setData] = useState(undefined);
+
+  useEffect(() => {
+    Index().then((result: History[]) => {
+      setData(result);
+    });
+  });
```

Comment:
no dependency array?

Reply:
does dependency array mean the second argument [] on this code?

``` tsx
  useEffect(() => {
    Index().then((result: History[]) => {
      setData(result);
    });
  }, []);
```

if yes then when using that it won't re-render after save button clicked

Reply:
empty dependency array means on first render.
so after save, call it again. so it will call twice. at start and after save. if not, then it will be called on every rerender

Reply:
I see, updated in https://github.com/RR-Ardhika/murojaah/pull/8/commits/1dd4063b3c9d30801121f492db49fa271de709dd, thank you